third-party/re2/re2-src/re2/nfa.cc

```diff
@@ -700,8 +700,8 @@ bool NFA<StrPiece>::Search(const StrPiece& text, const StrPiece& const_context,
     for (int i = 0; i < nsubmatch; i++)
       submatch[i].set_ptr_end(match_[2 * i], match_[2 * i + 1]);
     if (ExtraDebug)
-      fprintf(stderr, "match (%td,%td)\n",
-              match_[0] - btext_, match_[1] - btext_);
+      fprintf(stderr, "match (%td,%ld)\n",
+              (long)(match_[0] - btext_), (long)(match_[1] - btext_));
```

Comment:
Oh, wait, if it's a file offset and we're on a 32-bit machine, that will chop bits too.  I guess the best move is static_cast\<unsigned long long\> and use %llu in the printf.